pkg/ipam/allocator/multipool/allocator.go

```diff
@@ -23,16 +26,31 @@ type Allocator struct {
 }
 
 func (a *Allocator) Init(ctx context.Context, logger *slog.Logger, _ *metrics.Registry) error {
-	a.poolAlloc = NewPoolAllocator(logger)
+	a.poolAlloc = NewPoolAllocator(logger, nil)
 	a.logger = logger.With(subsysLogAttr...)
 	return nil
 }
 
-func (a *Allocator) Start(ctx context.Context, getterUpdater ipam.CiliumNodeGetterUpdater, _ *metrics.Registry) (allocator.NodeEventHandler, error) {
+func (a *Allocator) Start(ctx context.Context, getterUpdater ipam.CiliumNodeGetterUpdater, reg *metrics.Registry) (allocator.NodeEventHandler, error) {
+	var metricsIPAM metricsAPI
+	if operatorOption.Config.EnableMetrics {
+		triggerMetricsIPAM := ipamMetrics.NewTriggerMetricsIPAM(metrics.Namespace, "remaining_ips")
```

Comment:
These are not k8s trigger related metrics, so we should use a proper name like `remainingIPsMetric` or simply `multiPoolMetrics`.
 Moreover, there's no need to pass the metric name `remaining_ips` as a parameter, since it is just a constant string.